### PR TITLE
Replace UserInfo avatar with <MemberAvatar/> for fallback logic

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/11427

Even works on users who have naughty HTTP instead of MXC Avatars:
image

### CODE REVIEW

File: res/css/views/right_panel/_UserInfo.scss [hunk 1]
```diff
@@ -63,26 +63,30 @@ limitations under the License.
...
-    .mx_UserInfo_avatar > div > div {
+    .mx_UserInfo_avatar > div * {
         /* use padding-top instead of height to make this element square,
         as the % in padding is a % of the width (including margin,
         that's why we had to put the margin to center on a parent div),
         and not a % of the parent height. */
-        padding-top: 100%;
-        height: 0;
+        width: 30vh;
+        height: 30vh;
```
Comment (bwindels) [inline res/css/views/right_panel/_UserInfo.scss]: Does the avatar still grow with the width of the right panel with this?

Comment (t3chguy) [reply res/css/views/right_panel/_UserInfo.scss]: Oh right I didn't notice it grew between a min and max, interesting

Comment (bwindels) [reply res/css/views/right_panel/_UserInfo.scss]: Yeah, that's why I didn't go with MemberAvatar at the time, although that doesn't mean there isn't a way :)

Comment (t3chguy) [reply res/css/views/right_panel/_UserInfo.scss]: Arg I'm getting close 
image

File: src/components/views/right_panel/UserInfo.js [hunk 2]
```diff
@@ -1158,21 +1151,24 @@ const UserInfo = withLegacyMatrixClient(({matrixClient: cli, user, groupId, room
...
-    }
+    // const avatarUrl = user.getMxcAvatarUrl ? user.getMxcAvatarUrl() : user.avatarUrl;
+    const MemberAvatar = sdk.getComponent('avatars.MemberAvatar');
+    const avatarElement = (
+        <div className="mx_UserInfo_avatar">
+            <div>
+                <div>
+                    <MemberAvatar
+                        member={user}
+                        width={2 * 0.3 * window.innerHeight} // 2x@30vh
```
Comment (bwindels) [inline src/components/views/right_panel/UserInfo.js]: where does the 2x come from? The max-height in the css was 30vh fwiw...

Comment (bwindels) [reply src/components/views/right_panel/UserInfo.js]: I guess this also means this won't resize the avatar when resizing the window... which is OK I suppose.

File: res/css/views/right_panel/_UserInfo.scss [hunk 3]
```diff
@@ -77,12 +76,27 @@ limitations under the License.
         that's why we had to put the margin to center on a parent div),
```
Comment (bwindels) [inline res/css/views/right_panel/_UserInfo.scss]: is the padding + comment + max-width above still needed?

File: src/components/views/right_panel/UserInfo.js (hunk 2 above)
Comment (t3chguy) [reply src/components/views/right_panel/UserInfo.js]: It originally had 800x800px which I guess is not that bad, downloading a scaled image at exactly 30vh led to it rendering slightly blurry, so I opted to download at double-res

File: res/css/views/right_panel/_UserInfo.scss (hunk 3 above)
Comment (t3chguy) [reply res/css/views/right_panel/_UserInfo.scss]: Yes, I left the bare CSS required to make it work, underlying its doing the same trick as before but + position:relative with an absolutely positioned child which pins itself to fill the parent.

**Pull request merged**